app/src/main/java/app/foodfinderapp/MainActivity.kt

```diff
@@ -90,25 +91,33 @@ class MainActivity : AppCompatActivity() {
             }
         }
 
-        val restaurantRef = db.collection("restaurants")
-        restaurantRef.get()
-            .addOnSuccessListener { documents ->
-            for (document in documents) {
-                val restaurant = document.toObject(Restaurant::class.java)
-                restaurantList.add(restaurant)
+        restaurantService.getAllRestaurants(
```

Comment:
The MainActivity class is handling a lot (UI, navigation, and fetching data from Firestore). I moved the Firestore related code to it's own separate class (see restaurantService)

Reply:
It's definitely a good idea to move persistence (via Firestore) to another layer.  I recommend moving it to a ViewModel.  ViewModel is where state lives.